packages/client/hmi-client/src/components/documents/Document.vue

```diff
@@ -161,6 +199,31 @@ function highlightSearchTerms(text: string | undefined): string {
 	return text ?? '';
 }
 
+onMounted(async () => {
+	githubUrls.value = await getGithubUrls();
+});
+
+async function openCodeBrowser(url) {
+	isModalVisible.value = true;
+	chosenRepositoryName.value = url.full_name;
+	filesToSelect.value = await getGithubRepositoryContent(url.contents_url.slice(0, -8));
+}
+
+async function openCode(url) {
+	if (url.download_url === null) {
+		chosenRepositoryName.value = `${chosenRepositoryName.value}/${url.name}`;
+		filesToSelect.value = await getGithubRepositoryContent(url.url);
+		return;
+	}
+
+	emit(
+		'open-asset',
+		{ assetName: 'New file', assetType: ProjectAssetTypes.CODE },
+		// { assetName: url.name, assetId: url.name, assetType: ProjectAssetTypes.CODE },
```

Comment:
Why the comment line?

Reply:
That wouldn't open a tab properly since it's not registered as an asset in the resourceStore/resource sidebar yet. So for now the New File resource is opened and the code is pasted into it. I thought adding a new project asset within the project view like this (such as creating a code file) should be in a separate PR since it comes with its own challenges.